driver/src/main/java/oracle/nosql/driver/http/Client.java

```diff
@@ -1164,6 +1174,59 @@ public void enableRateLimiting(boolean enable, double usePercent) {
         }
     }
 
+    /**
+     * Get the proxy and kv versions from the response header.
+     * Update local values if different.
+     */
+    private void processVersionHeader(HttpHeaders headers) {
+        if (headers == null) {
+            return;
+        }
+        String versions = headers.get(PROXY_VERSION_HEADER);
+        if (versions == null) {
+            return;
+        }
+        /* versions string is space-separated k=v pairs */
+        String[] kvs = versions.split(" ");
+        if (kvs == null || kvs.length == 0) {
+            return;
+        }
+        for (int x=0; x<kvs.length; x++) {
+            String[] kv = kvs[x].split("=");
+            if (kv == null || kv.length != 2) {
+                continue;
+            }
+            if (kv[0].compareTo("proxy")==0) {
+                if (proxyVersion == null ||
+                    proxyVersion.compareTo(kv[1]) != 0) {
+                    proxyVersion = kv[1];
+                }
+            } else if (kv[0].compareTo("kv")==0) {
+                if (KVVersion == null ||
+                    KVVersion.compareTo(kv[1]) != 0) {
+                    KVVersion = kv[1];
+                }
+            }
+        }
+    }
+
+    /**
+     * @hidden
```

Comment:
I don't think the hidden tag is needed in this file since it's not included in javadoc